tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -137,6 +139,78 @@ def test_write_with_node_failures(self):
             self.run_validation()
         ctx.assert_actions_triggered()
 
+    @cluster(num_nodes=6, log_allow_list=CHAOS_LOG_ALLOW_LIST)
+    def test_write_with_node_decommissions(self):
+        self.start_producer()
+        produce_until_segments(
+            redpanda=self.redpanda,
+            topic=self.topic,
+            partition_idx=0,
+            count=20,
+        )
+
+        self.kafka_tools.alter_topic_config(
+            self.topic,
+            {
+                TopicSpec.PROPERTY_RETENTION_BYTES:
+                5 * EndToEndShadowIndexingTest.segment_size
+            },
+        )
+
+        with random_decommissions(
+                self.redpanda,
+                ActionConfig(
+                    cluster_start_lead_time_sec=20,
+                    min_time_between_actions_sec=5,
+                    max_time_between_actions_sec=10,
+                )) as ctx:
+            wait_for_segments_removal(redpanda=self.redpanda,
+                                      topic=self.topic,
+                                      partition_idx=0,
+                                      count=6)
+            objects = self.s3_client.list_objects(self.s3_bucket_name)
+            assert objects
+
+            self.start_consumer()
+            self.run_validation()
+        ctx.assert_actions_triggered()
+
+    @cluster(num_nodes=6)
+    def test_write_with_leadership_transfers(self):
```

Comment:
Why is the `test_write_with_*` tests necessary when the additions to franz_go_verifiable_test also do writes?

Reply:
The tests added to the franz-go ecosystem are are long running and take more time, so they are skipped in CI runs,they were used while doing manual testing, and once we have some kind of nightly run in place for these long running tests the distruptive tests can be added there.

The tests in e2e module such as `test_write_with_leadership_transfers` are fast and are part of the CI run, so they are kept separate. They are expected to run and pass with each CI build.

Reply:
Would be good to add a docstring to this comment to explain that it's purpose is to be a mini version of the more full-powered tests in scale_tests/

Reply:
added docstring to the test suite